File: ProAgil.Domain/Evento.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace ProAgil.Domain
{
    public class Evento
    {
        public int Id { get; set; }

        [Required(ErrorMessage = "This Field is Required!")]
        [DataType(DataType.DateTime, ErrorMessage = "This Date is invalid")]
        public DateTime  DataEvento { get; set; }
        [Required]
        public string  Local { get; set; } 
        [Required]
        [MinLength(5)]
        public string  Tema { get; set; }
        [Range(2,120000)]
        public int QtdPessoas { get; set; }
        [Phone]
        public string Telefone { get; set; }
        [EmailAddress]
        public string  Email { get; set; }
        public List<Lote> Lote { get; set; }
        public List<RedeSocial> RedeSocial { get; set; }
        public List<PalestranteEvento> PalestrantesEvento { get; set; }
        public string ImagemURL { get; set; }
    }
}

File: ProAgil-App/src/app/_models/Evento.ts
import { Lote } from './Lote';
import { RedeSocial } from './RedeSocial';
import { Palestrante } from './Palestrante';

export class Evento {
        constructor() { }
        id: number ;
        dataEvento: Date;
        local: string;
        tema: string;
        qtdPessoas: number;
        telefone: string;
        email: string;
        lotes: Lote[];
        redesociais: RedeSocial[];
        palestrantesEvento: Palestrante[];
        imagemURL: string;
} 